pandas/core/series.py

```diff
@@ -665,7 +666,20 @@ def __array__(self, result=None):
         """
         The array interface, return my values.
         """
-        return self.get_values()
+        # TODO: change the keyword name from result to dtype?
```

Comment:
Thoughts here? I think it would break people doing `np.asarray(ser, result=thing)`, but do people do that?  Does anyone know why it's called result in the first place?

Reply:
No idea. But maybe simply add dtype as second keyword, and for now keep result as third? The both positional and named argument would keep working. And we could also deprecate `result` then.

Reply:
I'm kinda thinking we just change this.

Users aren't supposed to call dunder methods directly right? And NumPy already filters out the bad keywords.

```python
In [10]: np.asarray(ser, dtype='int')
Out[10]: array([946681200000000000, 946767600000000000])

In [11]: np.asarray(ser, result='int')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-11-918c5400713e> in <module>
----> 1 np.asarray(ser, result='int')

TypeError: asarray() got an unexpected keyword argument 'result'
```

Reply:
Ah, yes, agree to directly change it then.